### PR TITLE
Terminate the threadpool executor when HTTPServer is destroyed.

### PR DESCRIPTION
**Pull request by fviale**

### CODE REVIEW

File: programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java [hunk 1]
```diff
@@ -122,12 +124,13 @@ public void stop() throws Exception {
         this.server.stop();
```
Comment (jrochas) [inline programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java:124]: Maybe you want to release the thread pool here as well

File: programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java [hunk 2]
```diff
@@ -122,12 +124,13 @@ public void stop() throws Exception {
...
     }
 
-    /** destroy the HTTP server, terminates the http server thread 
+    /** destroy the HTTP server, terminates the http server threads
      * 
      * @throws Exception If the HTTP server fails to stop
      */
     public void destroy() throws Exception {
         this.server.destroy();
+        this.threadPool.terminate();
```
Comment (jrochas) [inline programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java:133]: Does it matter when the termination of the thread pool is done? Before destroying the server maybe?

File: programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java (hunk 1 above)
Comment (fviale) [reply programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java:124]: It looks more coherent to me to shutdown the threadpool only when the server is destroyed.

File: programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java (hunk 2 above)
Comment (fviale) [reply programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java:133]: I would say explicitely after destroying the server, as the threadpool may be used during the destruction process...

File: programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java (hunk 1 above)
Comment (jrochas) [reply programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java:124]: ok

File: programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java (hunk 2 above)
Comment (jrochas) [reply programming-core/src/main/java/org/objectweb/proactive/core/httpserver/HTTPServer.java:133]: ok

**Review by jrochas — APPROVED** [review]

**Pull request merged**